ops/pebble.py

```diff
@@ -201,11 +204,11 @@ def __enter__(self) -> typing.IO[typing.AnyStr]: ...  # noqa
                            'kind': str,
                            'summary': str,
                            'status': str,
-                           'log': Optional[List[str]],
+                           'log': NotRequired[Optional[List[str]]],
                            'progress': _ProgressDict,
                            'spawn-time': str,
-                           'ready-time': str,
-                           'data': Optional[Dict[str, Any]]})
+                           'ready-time': NotRequired[Optional[str]],
+                           'data': NotRequired[Optional[Dict[str, Any]]]})
```

Comment:
Not a comment for this line, but it looks like `_ChangeDict` `tasks`, `err`, `ready_time`, and `data` should be `NotRequired`.

Reply:
Yes - I'd only done the ones that the tests were hitting, but indeed better to tighten then all up at once.